csv_permissions/evaluators.py

```diff
@@ -0,0 +1,76 @@
+from typing import Optional
+import warnings
+
+from csv_permissions.types import Evaluator
+from csv_permissions.types import UnresolvedEvaluator
+
+
+def make_evaluate_not_implemented(message) -> Evaluator:
+    """Creates a new evaluator that when invoked throws a NotImplementedError error
+    with a given message"""
+    def evaluate_not_implemented(user, obj=None) -> bool:
+        raise NotImplementedError(message)
+    return evaluate_not_implemented
+
+
+def resolve_validation_evaluator(details: UnresolvedEvaluator) -> Optional[Evaluator]:
+    """
+    This doesn't actually resolve to any evaluators, it just performs some validation
+    """
+    if details.model is None and not details.is_global:
+        raise ValueError("Permissions without Models must be global.")
+    return None
+
+
+def evaluate_all(user, obj=None) -> bool:
+    if obj is None:
+        raise ValueError("'all' cannot be used as a global permission.")
+    return True
+
+
+def resolve_all_evaluator(details: UnresolvedEvaluator) -> Optional[Evaluator]:
+    if details.evaluator_name == "all":
+        if details.is_global:
+            raise ValueError("'all' cannot be used as a global permission.")
+        return evaluate_all
+    return None
+
+
+def evaluate_yes(user, obj=None) -> bool:
+    if obj is not None:
+        raise ValueError("'yes' cannot be used as an object-level permission.")
+    return True
+
+
+def resolve_yes_evaluator(details: UnresolvedEvaluator) -> Optional[Evaluator]:
+    if details.evaluator_name == "yes":
+        if not details.is_global:
+            raise ValueError("'yes' cannot be used as an object-level permission.")
+        return evaluate_yes
+    return None
+
+
+def evaluate_no(user, obj=None) -> bool:
```

Comment:
for consistency with `yes` should we also accept `no` in the csv rather than just empty string? maybe it's more readable to be able to scan across a row and see yes,no,no rather than yes,,

Reply:
That used to be allowed, but it meant that there was two different ways to express the same thing.

I found it easier to scan big files if empty cells meant "no permissions".

Reply:
ahahahaha, and its not just me and dave want a "no" - ash the newcomer just left this msg in his training:

```
For the admin/customer permissions (the last couple of columns in the CSV), is there a reason we can't specify "no" and we're instead forced to just leave that cell blank?

I had "no" in them and couldn't figure out why it was throwing me errors, I ended up reading through the permission processing file to figure it out. (https://github.com/AllianceSoftware/django-csvpermissions/blob/master/csv_permissions/permissions.py#L91)
 

I know that in the training MD file, it does say "Blank cells are equivalent to that user type having no permission", but I interpreted that as "a cell could either be blank OR no" and I figured at the time that reading no's in the CSV would be easier, rather than intermittent blank cells.
```

so, really, "no" is what people _would_ be expecting 😆, in addition to being much more friendlier to `cat`

Reply:
okay, will change this.

@fanglong-alliance @davecoates this raises the question of what a blank cell means:

- fall back to "no" permissions
- "unspecified"
 
where this matters if you have multiple csv files and one is blank and the other is "yes", right now blank != yes and it throw an error but if we treat it as "unspecified" then the "yes" should win out and not throw an error

my inclination is the latter

Reply:
yes, i think that would be how i'd expect it to work. this way it also opens up for people to, if they really want, define a single row of permissions in a file w/o worrying it conflict w/ others

Reply:
implemented